src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -57,25 +66,30 @@ public void ReadFrame(bool server, out byte[] buffer)
                 throw new VirtualNetworkConnectionBroken();
             }
 
-            bool dequeueSucceeded = false;
             int remainingTries = 3;
             int backOffDelayMilliseconds = 2;
 
             do
```

Comment:
>do [](start = 12, length = 2)

It was here before, but I think the loop is unnecessary. Once you pass through semaphore, you should be able to dequeue a frame. 
Trying read three times could be just an attempt to deal with test timeouts.

Reply:
It's probably unnecessary.  I'm going to leave it for now.  One thing at a time :)